sympy/physics/mechanics/body.py

```diff
@@ -557,8 +553,361 @@ def dcm(self, body):
 
         """
 
-        if isinstance(body,  ReferenceFrame):
-            frame=body
-        elif isinstance(body, Body):
-            frame = body.frame
-        return self.frame.dcm(frame)
+        if isinstance(body, Body):
+            body = body.frame
+        return self.frame.dcm(body)
+
+    def orient_axis(self, parent, axis, angle):
+        """Sets the orientation of this body with respect to a parent body or
+        reference frame by rotating through an angle about an axis fixed
+        in the parent reference frame or body.
+
+        Parameters
+        ==========
+
+        parent : Body or ReferenceFrame
+            Body or Reference frame that this body will be rotated relative
+            to.
+        axis : Vector
+            Vector fixed in the parent frame about which this body is
+            rotated. It need not be a unit vector and the rotation follows the
+            right hand rule.
+        angle : sympifiable
+            Angle in radians by which the frame is to be rotated.
+
+        Warns
+        ======
+
+        UserWarning
+            If the orientation creates a kinematic loop.
+
+        Examples
+        ========
+
+        Setup variables for the examples:
+
+        >>> from sympy import symbols
+        >>> from sympy.physics.mechanics import Body
+        >>> q1 = symbols('q1')
+        >>> A = Body('A')
+        >>> B = Body('B')
+        >>> B.orient_axis(A, A.x, q1)
+
+        The ``orient_axis()`` method generates a direction cosine matrix and
+        its transpose which defines the orientation of B relative to A and vice
+        versa. Once orient is called, ``dcm()`` outputs the appropriate
+        direction cosine matrix:
+
+        >>> B.dcm(A)
+        Matrix([
+        [1,       0,      0],
+        [0,  cos(q1), sin(q1)],
+        [0, -sin(q1), cos(q1)]])
+        >>> A.dcm(B)
+        Matrix([
+        [1,       0,        0],
+        [0, cos(q1), -sin(q1)],
+        [0, sin(q1),  cos(q1)]])
+
+        The following two lines show that the sense of the rotation can be
+        defined by negating the vector direction or the angle. Both lines
+        produce the same result.
+
+        >>> B.orient_axis(A, -A.x, q1)
+        >>> B.orient_axis(A, A.x, -q1)
+
+        """
```

Comment:
I don't think we need such detailed examples here as they are already present in original function so it would be wise to add a simple example here and everywhere else and add a see also section which points to original orient function.

Reply:
Same goes to docstrings , don't elaborate as in original just the necessary part.